src/python/pants/bsp/goal.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import logging
+import os
+import sys
+
+from pants.base.exiter import ExitCode
+from pants.base.specs import Specs
+from pants.bsp.protocol import BSPConnection
+from pants.build_graph.build_configuration import BuildConfiguration
+from pants.engine.unions import UnionMembership
+from pants.goal.builtin_goal import BuiltinGoal
+from pants.init.engine_initializer import GraphSession
+from pants.option.options import Options
+
+_logger = logging.getLogger(__name__)
+
+
+class BSPGoal(BuiltinGoal):
+    name = "bsp"
+    help = 'Run server for Build Server Protocol ("BSP").'
```

Comment:
Worth including a link to the homepage in here?

Reply:
Will add.

Should this also be named "experimental-bsp" to users don't try and use it?

Reply:
Done (for both).

Reply:
> Should this also be named "experimental-bsp" to users don't try and use it?

Yea, probably... since it's not in a backend, it will be visible by default for all users.